### DISCUSSION TITLE
Help with avoiding DocumentLayoutException on table with user input

### DISCUSSION THREAD

**Discussion by BeneHenke** [QuestPDF/QuestPDF#980] [Q&A]
Hello,

I am working on an application that is supposed to render PDF files based on user input. Part of that is the users being able to define columns sizes of a table themselves. Is it possible to catch if a user has put in impossible numbers and adjust them to the space that is available?
I have tried following the example for the dynamic components and it works for single columns but doesn't seem to work with multicolumn elements, probably due to them being composed multiple times and the State not adjusting to that.

Thanks

**Comment by BeneHenke** [comment]
I introduced a variable that saves the current page size and update it every time I change the page size. Like this I can use the normal table elements. There is probably a better way but it works like this.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Glad you were able to find a working solution. We are closing this thread as resolved.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
